apps/omg/lib/omg/state/transaction/fee_token_claim.ex

```diff
@@ -114,14 +136,30 @@ defimpl OMG.State.Transaction.Protocol, for: OMG.State.Transaction.FeeTokenClaim
   def get_inputs(%Transaction.FeeTokenClaim{}), do: []
 
   @doc """
-  Fee claiming transaction is not used to transfer funds
+  Tells whether Fee claiming transaction is valid
   """
-  @spec valid?(Transaction.FeeTokenClaim.t(), Transaction.Signed.t()) :: {:error, atom()}
-  def valid?(%Transaction.FeeTokenClaim{}, _signed_tx), do: {:error, :not_implemented}
+  @spec valid?(Transaction.FeeTokenClaim.t(), Transaction.Signed.t()) ::
+          {:error, :wrong_number_of_fee_outputs | :fee_output_amount_has_to_be_positive}
+  def valid?(%Transaction.FeeTokenClaim{} = fee_tx, _signed_tx) do
+    # we're able to check structure validity => single output with amount > 0
+    with outputs = Transaction.get_outputs(fee_tx),
+         true <- length(outputs) == 1 || {:error, :wrong_number_of_fee_outputs},
+         [output] = outputs,
+         true <- output.amount > 0 || {:error, :fee_output_amount_has_to_be_positive},
+         do: true
+  end
 
   @doc """
   Fee claiming transaction is not used to transfer funds
```

Comment:
this requires a comment on the `outputs` input - that it is expected to be prepared in line with the protocol interface. So we're having the `FeeTokenClaim` spend pseudo-outputs based on the `fees_paid`.

These explanations should be split between this `@doc` here and the spot where it's called